android/src/main/java/com/swmansion/gesturehandler/core/RotationGestureDetector.kt

```diff
@@ -85,50 +120,109 @@ class RotationGestureDetector(private val gestureListener: OnRotationGestureList
     }
   }
 
+  /**
+   * Gesture may be paused when second Pointer lifts
+   * with secondPointerLiftFinishesGesture set to `false`
+   *
+   * Then Detector waits for a new second Pointer to arrive to continue Handling
+   * (last Pointer Lifting still finishes the Gesture)
+   *
+   * @see secondPointerLiftFinishesGesture
+   * @see pause
+   * @see unpause
+   */
+  private var isPaused = false
+  private fun pause() {
+    localLog(".pause() requested", true)
+    if (!isPaused) {
+      isPaused = true
+      localLog(".pause() fulfilled")
+    }
+  }
+  private fun unpause(eventAngle: Double) {
+    localLog(".unpause(eventAngle: $eventAngle) requested", true)
+    if (isPaused) {
+      previousAngle = eventAngle
+      isPaused = false
+      localLog(".unpause(eventAngle: $eventAngle) fulfilled")
+    }
+  }
+
   private fun finish() {
     if (isInProgress) {
+      isPaused = false
       isInProgress = false
       gestureListener?.onRotationEnd(this)
     }
   }
 
   fun onTouchEvent(event: MotionEvent): Boolean {
+    fun getLogInfo(): String {
+      val firstPointerIndex = event.findPointerIndex(pointerIds[0])
+      val secondPointerIndex = event.findPointerIndex(pointerIds[1])
+
+      return """
+        pointerIds[0]: ${pointerIds[0]}, pointerIds[1]: ${pointerIds[1]}
+        firstPointerIndex: $firstPointerIndex, secondPointerIndex: $secondPointerIndex
+        isInProgress: $isInProgress, isPaused: $isPaused
+        action: ${event.action}, actionIndex: ${event.actionIndex} pointerCount: ${event.pointerCount}
+        rotation: $rotation, previousAngle: $previousAngle
+        event: $event
+      """.trimIndent()
+    }
+    localLog("onTouchEvent START | ${getLogInfo()}", true)
+
     when (event.actionMasked) {
       MotionEvent.ACTION_DOWN -> {
+        localLog("ActionDown START | ${getLogInfo()}")
         isInProgress = false
         pointerIds[0] = event.getPointerId(event.actionIndex)
         pointerIds[1] = MotionEvent.INVALID_POINTER_ID
+        localLog("ActionDown END | ${getLogInfo()}")
       }
-      MotionEvent.ACTION_POINTER_DOWN -> if (!isInProgress) {
-        pointerIds[1] = event.getPointerId(event.actionIndex)
-        isInProgress = true
-        previousTime = event.eventTime
-        previousAngle = Double.NaN
-        updateCurrent(event)
-        gestureListener?.onRotationBegin(this)
+      MotionEvent.ACTION_POINTER_DOWN -> {
+        localLog("ActionPointerDown START | ${getLogInfo()}")
+        if (!isInProgress) {
+          pointerIds[1] = event.getPointerId(event.actionIndex)
+          isInProgress = true
+          previousTime = event.eventTime
+          previousAngle = Double.NaN
+          updateCurrent(event)
+          gestureListener?.onRotationBegin(this)
+        }
```

Comment:
Similarly here, the pointer id should be updated

```suggestion
        if (!isInProgress || isPaused) {
          pointerIds[1] = event.getPointerId(event.actionIndex)
          updateCurrent(event)
        }

        if (!isInProgress) {
          isInProgress = true
          previousTime = event.eventTime
          previousAngle = Double.NaN
          gestureListener?.onRotationBegin(this)
        }
```

Reply:
According to suggested [here](https://github.com/software-mansion/react-native-gesture-handler/pull/3057#discussion_r1791654362) – replacing `MotionEvent.INVALID_POINTER_ID`

Reply:
And here, if you [shift the pointers](https://github.com/software-mansion/react-native-gesture-handler/pull/3057/files/fec2867e424b465d6d28c025e0c5a436f58a4b9f#r1793052901), you don't need to replace the `MotionEvent.INVALID_POINTER_ID` but simply assume that `pointerIds[1]` should be set (like it's now).